Backend/src/server.ts

```diff
@@ -8,19 +8,34 @@ import cors from 'cors';
 import schema from './schema';
 import { ApiError } from './customErrors/apiError';
 import authMiddleware from "./middlewares/basicAuth";
+import initPassport from './middlewares/passportOauth';
+import passport from 'passport';
 
+initPassport();
 const app = express();
 
+app.use(passport.initialize());
+
+// Keept out because typeScript is angry
+const params = { scope: 'openid email', accessType: 'offline', prompt: 'consent' }
```

Comment:
can you explain "scope". What does it ask for here? Openid is the token? 

accessType: 'offline' ? :)

Reply:
scope is what we want - we want the openid primarily. I'm unsure it this is actually needed as what we ask for is (I think) always included. This can be testet further. This is just first working version I made. 
accessType 'offline' is needed in order to get the refresh token, otherwise we won't get that

Reply:
I'm guessing that this is the *scope of information that we request*? (As explained in the videos we all saw)

This is what users will be presented by - that we request their email - for use with our application.

It is important to keep it short. "Woke" users won't give away all their data, and therefore wont use the app.
Also, a breach will only give away an email address, not **everything else**.

We can create multiple scopes for further requests, e.g. adding user addresses for shipping. (which will not be needed for our app)

these scopes are best kept apart.

Reply:
> I'm guessing that this is the scope of information that we request? (As explained in the videos we all saw)

Yeah, exactly. It is what you're asking for, from Google. What Permissions are you asking the User to confirm, and what info are you requesting.

Reply:
> This is what users will be presented by - that we request their email - for use with our application.
> 
> It is important to keep it short. "Woke" users won't give away all their data, and therefore wont use the app.
> Also, a breach will only give away an email address, not everything else.
> 
> We can create multiple scopes for further requests, e.g. adding user addresses for shipping. (which will not be needed for our app)
> 
> these scopes are best kept apart.

Det er lige præcis _**Incremental Authorization**_ du beskriver der :) 
https://developers.google.com/identity/sign-in/web/incremental-auth